src/spider/worker/WorkerClient.cpp

```diff
@@ -69,21 +91,23 @@ auto WorkerClient::get_next_task(std::optional<boost::uuids::uuid> const& fail_t
     std::default_random_engine rng{random_device()};
     std::ranges::shuffle(schedulers, rng);
 
-    std::vector<boost::asio::ip::tcp::endpoint> endpoints;
-    std::ranges::transform(
-            schedulers,
-            std::back_inserter(endpoints),
-            [](core::Scheduler const& scheduler) {
-                return boost::asio::ip::tcp::endpoint{
-                        boost::asio::ip::make_address(scheduler.get_addr()),
-                        static_cast<unsigned short>(scheduler.get_port())
-                };
-            }
-    );
     try {
         // Create socket to scheduler
         boost::asio::io_context context;
         boost::asio::ip::tcp::socket socket(context);
+
+        boost::asio::ip::tcp::resolver resolver(context);
+        std::vector<boost::asio::ip::tcp::endpoint> endpoints;
+        for (auto const& scheduler : schedulers) {
+            std::vector<boost::asio::ip::tcp::endpoint> const resolved_endpoints
+                    = resolve_hostname(context, scheduler.get_addr(), scheduler.get_port());
+            endpoints.insert(
+                    endpoints.end(),
+                    resolved_endpoints.begin(),
+                    resolved_endpoints.end()
```

Comment:
```suggestion
                    endpoints.cend(),
                    resolved_endpoints.cbegin(),
                    resolved_endpoints.cend()
```
Nit: Use const version iterator when possible.